### PR TITLE
Alias resolution for accountHttp

### LINKED ISSUE #499: Return namespaces names from namespace and mosaic endpoint.
it's big overhead return namespaces names from the sdk?

`NamespaceHttp(http.nodeUrl).getNamespacesFromAccount`
`AccountHttp(http.nodeUrl).getAccountInfo`

because in client apps development, we need make extra call for the names endpoint to get namespace name

@evias @decentraliser 

related topic
https://nem2.slack.com/archives/CEZKUE4KB/p1583900407059200

### PR DESCRIPTION
**Pull request by rg911**
- Added `AccountService` with 2 new methods: `accountInfoWithResolvedMosaic` and `accountNamespacesWithName`
- Added types to hold `NamespaceName` for `AccountInfo` and `NamespaceInfo`
- Added tests

Fixed #499

### CODE REVIEW

File: test/service/AccountServce.spec.ts [hunk 1]
```diff
@@ -0,0 +1,190 @@
+/*
```
Comment (dgarcia360) [inline test/service/AccountServce.spec.ts:1]: File name (AccountService.spec.ts)

File: src/service/AccountService.ts [hunk 2]
```diff
@@ -0,0 +1,137 @@
...
+            }
+        });
+    }
+
+    /**
+     * Get distince list of namespacesIds from list of accountInfos
+     * @param accountInfo List of account info
+     * @returns {NamespaceId[]}
+     */
+    private getDistinctNamespaceIdFromAccountInfors(accountInfo: AccountInfo[]): NamespaceId[] {
```
Comment (dgarcia360) [inline src/service/AccountService.ts:124]: typo "Infos"

File: src/service/AccountService.ts [hunk 3]
```diff
@@ -0,0 +1,137 @@
...
+                            return DtoMapping.assign(info, { namespaceName: name?.name });
+                        });
+                    }),
+                );
+            }),
+        );
+    }
+
+    /**
+     * Resolve mosaics provides namespace names
```
Comment (dgarcia360) [inline src/service/AccountService.ts:100]: Resolve mosaics provided namespace names

File: src/service/AccountService.ts [hunk 4]
```diff
@@ -0,0 +1,137 @@
...
+                    return DtoMapping.assign(mosaic, { namespaceName: name });
+                } else {
+                    return mosaic as ResolvedMosaic;
+                }
+            }
+        });
+    }
+
+    /**
+     * Get distince list of namespacesIds from list of accountInfos
```
Comment (dgarcia360) [inline src/service/AccountService.ts:120]: distince -> distinct

File: src/service/AccountService.ts (hunk 2 above)
Comment (dgarcia360) [inline src/service/AccountService.ts:124]: I'm not sure to understand what this function does, could you please give me some context where could be used?

Comment (rg911) [reply src/service/AccountService.ts:124]: To reduce the rest calls. Get a distinct list of namespaceIds and with 1 rest call to get all names. Then map the resolved names back to models.

File: src/service/AccountService.ts [hunk 5]
```diff
@@ -0,0 +1,137 @@
...
+            }
+        });
+    }
+
+    /**
+     * Get distinct list of namespacesIds from list of accountInfos
+     * @param accountInfo List of account infos
+     * @returns {NamespaceId[]}
+     */
+    private getDistinctNamespaceIdFromAccountInfors(accountInfo: AccountInfo[]): NamespaceId[] {
```
Comment (dgarcia360) [inline src/service/AccountService.ts:124]: AccountInfors -> AccountInfos

File: src/service/AccountService.ts (hunk 2 above)
Comment (dgarcia360) [reply src/service/AccountService.ts:124]: I see, I haven't seen that's defined as private 👍

**Pull request merged**